execution/process.go

```diff
@@ -261,5 +303,87 @@ func (p *ProcessByKey) ReadyForMore(ctx context.Context, tx storage.StateTransac
 }
 
 func (p *ProcessByKey) Close(ctx context.Context, storage storage.Storage) error {
+	p.garbageCollectorCtxCancel()
+	err := <-p.garbageCollectorCloseErrChan
+	if err == context.Canceled || err == context.DeadlineExceeded {
+	} else if err != nil {
+		return errors.Wrap(err, "couldn't stop garbage collector")
+	}
+
 	return nil
 }
+
+type GarbageCollectorInfo struct {
+	garbageCollectorBoundary int // number of seconds subtracted from watermark to create boundary for garbage collection
```

Comment:
If you're operating on duration values, please use durations, not int of seconds.